homeassistant/components/myuplink/number.py

```diff
@@ -0,0 +1,132 @@
+            key="degree_minutes",
+            icon="mdi:thermometer-lines",
+            native_unit_of_measurement="DM",
+        ),
+    },
+}
+
+
+def get_description(device_point: DevicePoint) -> NumberEntityDescription | None:
+    """Get description for a device point.
+
+    Priorities:
+    1. Category specific prefix e.g "NIBEF"
+    2. Global parameter_unit e.g. "DM"
+    3. Default to None
+    """
+    prefix, _, _ = device_point.category.partition(" ")
+    description = CATEGORY_BASED_DESCRIPTIONS.get(prefix, {}).get(
+        device_point.parameter_id
+    )
+
+    if description is None:
+        description = DEVICE_POINT_UNIT_DESCRIPTIONS.get(device_point.parameter_unit)
+
+    return description
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up myUplink number."""
+    entities: list[NumberEntity] = []
+    coordinator: MyUplinkDataCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    # Setup device point number entities
+    for device_id, point_data in coordinator.data.points.items():
+        for point_id, device_point in point_data.items():
+            description = get_description(device_point)
+            if find_matching_platform(device_point, description) == Platform.NUMBER:
+                entities.append(
+                    MyUplinkNumber(
+                        coordinator=coordinator,
+                        device_id=device_id,
+                        device_point=device_point,
+                        entity_description=description,
+                        unique_id_suffix=point_id,
+                    )
+                )
+
+    async_add_entities(entities)
+
+
+class MyUplinkNumber(MyUplinkEntity, NumberEntity):
+    """Representation of a myUplink number entity."""
+
+    def __init__(
+        self,
+        coordinator: MyUplinkDataCoordinator,
+        device_id: str,
+        device_point: DevicePoint,
+        entity_description: NumberEntityDescription | None,
+        unique_id_suffix: str,
+    ) -> None:
+        """Initialize the number."""
+        super().__init__(
+            coordinator=coordinator,
+            device_id=device_id,
+            unique_id_suffix=unique_id_suffix,
+        )
+
+        # Internal properties
+        self.point_id = device_point.parameter_id
+        self._attr_name = device_point.parameter_name
+        self._attr_native_min_value = (
+            device_point.raw["minValue"] if device_point.raw["minValue"] else -30000
+        ) * float(device_point.raw.get("scaleValue", 1))
+        self._attr_native_max_value = (
+            device_point.raw["maxValue"] if device_point.raw["maxValue"] else 30000
+        ) * float(device_point.raw.get("scaleValue", 1))
+        self._attr_step_value = device_point.raw.get("stepValue", 20)
+        if entity_description is not None:
+            self.entity_description = entity_description
```

Comment:
We should not add entities without an entity description. Can this even happen?

Reply:
Yes, it can happen. In fact it is common in sensor platform. The API delivers some 60 device points for e.g a Nibe F730. It is not uncommon that we can't identify the optimal device description from the meta-data. Then we create the entity without description. The name is most likely enough for the nerd-user to understand what the data is representing. But the names are not stable enough so we have chosen not to use that for selecting platform or device_class.
If a device_point has the attributes maxValue and minValue set and writable is true it is best represented in HA by a numeric entity even if we cannot assign an optimal device_class or icon.